src/client/process.ts

```diff
@@ -117,18 +117,35 @@ export class PowerShellProcess {
         return this.sessionDetails
     }
 
-    public async showTerminalIfVisible() {
+    public async showTerminalIfNotVisible() {
         if (this.consoleTerminal) {
             const winid: number = await vscode.workspace.nvim.eval(`bufwinid(${this.consoleTerminal.bufnr})`) as number;
 
-            // If winid is -1, it means the window is not visible/is hidden.
-            if (winid > -1) {
-                this.consoleTerminal.show(!this.config.integratedConsole.focusConsoleOnExecute);
+          // show terminal if hidded when running F5/F8
+            if (winid == -1) {
+                this.consoleTerminal.show();
+            }
+
+            // this will move cursor to consol after F5/F8 if focusConsoleOnExecute is true
+            if (this.config.integratedConsole.focusConsoleOnExecute) {
+               this.consoleTerminal.show();
             }
         }
     }
+    
+    public async showTerminal() {this.consoleTerminal.show();}
```

Comment:
```suggestion
    public showTerminal() {
        this.consoleTerminal.show();
    }
```
if you don't use await, the method shouldn't be marked async and also whitespace is nice :)